tools/importTool/src/main/scala/AkkaFileReaderWithActor.scala

```diff
@@ -0,0 +1,138 @@
+import akka.http.scaladsl.model.{HttpEntity, HttpMethods, HttpRequest, HttpResponse, _}
+import akka.pattern.ask
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Framing, Sink}
+import akka.util.{ByteString, Timeout}
+
+import scala.concurrent.duration.{FiniteDuration, _}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+case class ActorInput(inputUrl: String, outputFormat:String, cluster:String)
+case class TripleWithKey(subject: String, triple: String)
+
+
+class AkkaFileReaderWithActor(inputUrl:String, format:String, cluster:String) extends Actor {
+
+  import system.dispatcher
+  implicit val system = ActorSystem("app")
+  implicit val scheduler = system.scheduler
+  implicit val timeout = Timeout(5.seconds)
+
+  implicit val mat = ActorMaterializer()
+  val bytesAccumulatorActor = this.system.actorOf(Props(new BytesAccumulatorActor()), name = "offsetActor")
+  context.watch(bytesAccumulatorActor)
+
+  override def receive: Receive = {
+    case ActorInput => {
+      println("Starting flow.....")
+      readAndImportFile(inputUrl, format, cluster)
+    }
+    case Terminated(bytesAccumulatorActor) => {
+      println("BytesAccumulatorActor has been crashed, going to restart the flow")
+      readAndImportFile(inputUrl, format, cluster)
+
+    }
+  }
+
+  def readFileFromServer(inputUrl:String) = {
+    println("Going to send get request to inputUrl=" + inputUrl)
+      val future = bytesAccumulatorActor ? LastOffset
+      future.flatMap(x => {
+        val ranges: String = "bytes=" + x + "-"
+        println("Resending get request for ranges=" + ranges)
+        val request = Get(inputUrl).addHeader(RawHeader.apply("Range", ranges))
+        Http().singleRequest(request)
+      })
+  }
+
+  def readAndImportFile(inputUrl:String, format:String, cluster:String)= {
+    val httpResponse = readFileFromServer(inputUrl)
+    httpResponse.onComplete(
+      {
+        case Success(r) if  r.status.isSuccess() => println("Get file from server successfully")
+        case Success(r) if  r.status.isFailure() => println("Failed to read file, got status code:" + r.status)
+        case Failure(e) => {
+          println("Got a failure while reading the file, ", e.getMessage)
+          system.scheduler.scheduleOnce(7000.milliseconds, self, ActorInput)
+        }})
+    httpResponse.foreach({ response =>
+      println("Get response status=" + response.status)
+      val contentLengthHeader = response.headers.find(_.name == "Content-Range").map(_.value).getOrElse("")
+      val fileSize = contentLengthHeader.split("/")(1).toLong + 1
+      bytesAccumulatorActor ! FileSize(fileSize)
+      response.entity.withSizeLimit(fileSize).dataBytes
+        .via(Framing.delimiter(ByteString(System.lineSeparator()), 100000, true))
+        .map(_.utf8String)
+        .map(getTripleKey)
+        .sliding(2, 1)
+        .splitAfter(pair => (pair.head, pair.last) match {
```

Comment:
This is nicer: `.splitAfter(pair => pair.head.subject != pair.last.subject)`

Reply:
done